eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Certs.hs

```diff
@@ -213,59 +223,76 @@ conwayCertsTransition = do
-      -- because if there are no proposals to vote on , there will be no votes either.
-      let numDormantEpochs = certState' ^. certVStateL . vsNumDormantEpochsL
-          updateVSDReps vsDReps =
-            Map.foldlWithKey'
-              ( \dreps voter _ -> case voter of
-                  DRepVoter cred ->
-                    Map.adjust
-                      (drepExpiryL .~ computeDRepExpiry drepActivity currentEpoch numDormantEpochs)
-                      cred
-                      dreps
-                  _ -> dreps
-              )
-              vsDReps
-              (unVotingProcedures $ tx ^. bodyTxL . votingProceduresTxBodyL)
-
-      -- Final CertState with updates to DRep expiry based on new proposals and votes on existing proposals
-      let certStateWithDRepExpiryUpdated = certState' & certVStateL . vsDRepsL %~ updateVSDReps
-          dState = certStateWithDRepExpiryUpdated ^. certDStateL
-          withdrawals = tx ^. bodyTxL . withdrawalsTxBodyL
-
-      -- Validate withdrawals and rewards and drain withdrawals
-      failOnJust
-        (withdrawalsThatDoNotDrainAccounts withdrawals network (dState ^. accountsL))
-        WithdrawalsNotInRewardsCERTS
-
-      pure $ certStateWithDRepExpiryUpdated & certDStateL . accountsL %~ drainAccounts withdrawals
+    Empty ->
+      if hardforkConwayMoveWithdrawalsAndDRepChecksToLedgerRule $ pp ^. ppProtocolVersionL
+        then pure certState
+        else do
+          network <- liftSTS $ asks networkId
+          let accounts = certState ^. certDStateL . accountsL
+              withdrawals = tx ^. bodyTxL . withdrawalsTxBodyL
+          failOnJust
+            (withdrawalsThatDoNotDrainAccounts withdrawals network accounts)
+            ( \(invalid, incomplete) ->
+                WithdrawalsNotInRewardsCERTS $ Withdrawals $ unWithdrawals invalid <> unWithdrawals incomplete
+            )
+          pure $
+            certState
+              & updateDormantDRepExpiries tx currentEpoch
+              & updateVotingDRepExpiries tx currentEpoch (pp ^. ppDRepActivityL)
+              & certDStateL . accountsL %~ drainAccounts withdrawals
     gamma :|> txCert -> do
       certState' <-
         trans @(ConwayCERTS era) $ TRC (env, certState, gamma)
       trans @(EraRule "CERT" era) $
         TRC (CertEnv pp currentEpoch committee committeeProposals, certState', txCert)
 
+-- | If there is a new governance proposal to vote on in this transaction,
+-- AND the number of dormant-epochs recorded is greater than zero, we bump
+-- the expiry for all DReps by the number of dormant epochs, and reset the
+-- counter to zero.
+--
+-- It does not matter that this is called _before_ the GOV rule in LEDGER, even
+-- though we cannot validate any governance proposal here, since the entire
+-- transaction will fail if the proposal is not accepted in GOV, and so will
+-- this expiry bump done here. It will be discarded.
```

Comment:
It is unclear what this "It" that will be discarded means, so I think it is better to remove it (pun intended), since explanation above is clear enough.
```suggestion
-- this expiry bump done here.
```